common/resilient/http.go

```diff
@@ -0,0 +1,72 @@
+package resilient
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	retry "github.com/avast/retry-go"
+)
+
+type WrappedHttpClient struct {
+	underlying HttpClient
+	opts       []retry.Option
+}
+
+type HttpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func NewHttpClient(opts ...retry.Option) *WrappedHttpClient {
+	return WrapHttpClient(&http.Client{}, opts...)
+}
+
+func WrapHttpClient(client HttpClient, opts ...retry.Option) *WrappedHttpClient {
+	return &WrappedHttpClient{
+		underlying: client,
+		opts:       opts,
+	}
+}
+
+func (c *WrappedHttpClient) Do(req *http.Request) (*http.Response, error) {
```

Comment:
it's quite funny that in other methods you used named return params even when they are not needed. For this method, named param would be desirable but you didnt used them.

IMO you should use here the named params and drop them for other methods because they are not needed at all and they can only be misleading. 

```go
func (c *WrappedHttpClient) Do(req *http.Request) (resp *http.Response, err error)
func (c *WrappedHttpClient) Get(url string) (*http.Response, error)
func (c *WrappedHttpClient) Post(url, contentType string, body io.Reader) (*http.Response, error)
func (c *WrappedHttpClient) PostForm(url string, data url.Values) (*http.Response, error)
func (c *WrappedHttpClient) Head(url string) (*http.Response, error)
```

Reply:
The reason is quite simple. I copied public interface of http.Client to be compliant. Unfortunately I couldn't extend http.Client as all those additional methods use `Do` under the hood, so they would use `http.Client`s implementation, not mine.

Reply:
I've changed the signatures